test/Dialect/FIRRTL/annotations.fir

```diff
@@ -317,3 +326,92 @@ circuit Foo: %[[{"target": "Foo|Foo>_T_0", "class": "firrtl.transforms.DontTouch
         read-latency => 0
         write-latency => 1
         read-under-write => undefined
+
+; // -----
+
+; --------------------------------------------------------------------------------
+; SiFive-custom annotations related to the GrandCentral utility.  These
+; annotations do not conform to standard SingleTarget or NoTarget format and
+; need to be manually split up.
+; --------------------------------------------------------------------------------
+
+; Test sifive.enterprise.grandcentral.DataTapsAnnotation
+circuit GCTDataTap : %[
+[
+  {
+    "class": "sifive.enterprise.grandcentral.DataTapsAnnotation",
+    "blackBox": "~GCTDataTap|DataTap",
+    "keys": [
+      {
+        "class": "sifive.enterprise.grandcentral.ReferenceDataTapKey",
+        "source": "~GCTDataTap|GCTDataTap>r",
+        "portName": "~GCTDataTap|DataTap>_0"
+      }
+    ]
+  }
+]]
+  extmodule DataTap :
+    output _0 : UInt<1>
+
+    defname = DataTap
+
+
+  module GCTDataTap :
+    input clock : Clock
+    input reset : UInt<1>
+    input a : UInt<1>
+    output b : UInt<1>
+
+    reg r : UInt<1>, clock
+    r <= a
+    b <= r
+    wire cloneType : UInt<1>
+    inst DataTap of DataTap
+    DataTap._0 is invalid
+    cloneType <= DataTap._0
+
+    ; CHECK-LABEL: firrtl.circuit "GCTDataTap"
+    ; CHECK: firrtl.extmodule @DataTap
+    ; CHECK-SAME: %_0: [[A:.+]] {firrtl.annotations = [{class = "sifive.enterprise.grandcentral.ReferenceDataTapKey", hash = [[HASH:.+]] : i64}]}
+    ; CHECK-SAME: annotations = [{class = "sifive.enterprise.grandcentral.DataTapsAnnotation"}]
+
+    ; CHECK: firrtl.module @GCTDataTap
+    ; CHECK: firrtl.reg
+    ; CHECK-SAME: annotations = [{class = "sifive.enterprise.grandcentral.ReferenceDataTapKey", hash = [[HASH]] : i64}]
```

Comment:
Is the 64 bit hash strong enough to identify an annotation with sufficiently low chance of collisions? Should this maybe go through a uniquing process, or just allocate from a global annotation ID counter?

Reply:
Great idea. Yeah, this hash isn't a great approach and collisions would be really bad. In the rebase, I switched to a global ID tracked in the FIRRTL parser state.

Reply:
That sounds great! :tada: